pysat/tests/test_utils.py

```diff
@@ -466,3 +468,47 @@ def test_netcdf_attribute_override(self):
         # custom attribute correctly read from file
         assert meta.bespoke
 
+
+class TestBasicNetCDF4xarray():
+    def setup(self):
+        """Runs before every method to create a clean testing setup."""
+        # store current pysat directory
+        self.data_path = pysat.data_dir
+
+        # create temporary directory
+        dir_name = tempfile.mkdtemp()
+        pysat.utils.set_data_dir(dir_name, store=False)
+
+        self.testInst = pysat.Instrument(platform='pysat',
+                                         name='testing_xarray',
+                                         sat_id='100',
+                                         clean_level='clean')
+        self.testInst.pandas_format = False
+
+        # create testing directory
+        prep_dir(self.testInst)
+
+    def teardown(self):
+        """Runs after every method to clean up previous testing."""
+        remove_files(self.testInst)
+        try:
+            pysat.utils.set_data_dir(self.data_path, store=False)
+        except:
+            pass
```

Comment:
@rstoneback, do we need the try: except: here and above?  Coveralls indicates that the except is not being run

Reply:
Not sure why that is there. set_data_dir does throw an error if a path is invalid. 
Seems reasonable to keep a valid path check in there so we don't end up in a weird state. Especially if we aren't running into it already on coveralls.